file_system.go

```diff
@@ -202,17 +243,10 @@ func extractDirName(msg string) (string, error) {
 	return strings.Replace(msg[openQuote+1:closeQuote], `""`, `"`, -1), nil
 }
 
-func (c *Client) controlStringList(f string, args ...interface{}) ([]string, error) {
-	pconn, err := c.getIdleConn()
-	if err != nil {
-		return nil, err
-	}
-
-	defer c.returnConn(pconn)
-
+func (c *Client) controlStringList(pconn *persistentConn, f string, args ...interface{}) ([]string, error) {
```

Comment:
Mostly because I needed to access pconn.mlsdNotSupported in ReadDir for the connection. 
Once I did getIdleConn() in ReadDir() I should also pass it to the underlying functions.